src/microbe_stage/systems/MicrobeAISystem.cs

```diff
@@ -773,25 +787,42 @@ private void PursueAndConsumeChunks(ref WorldPosition position, ref MicrobeAI ai
     }
 
     private void FleeFromPredators(ref WorldPosition position, ref MicrobeAI ai, ref MicrobeControl control,
-        ref OrganelleContainer organelles, CompoundBag ourCompounds, in Entity entity, Vector3 predatorLocation,
-        float speciesFocus, float speciesActivity, float speciesAggression, float speciesFear, float strain,
-        Random random)
+        ref OrganelleContainer organelles, ref CompoundStorage compoundStorage, in Entity entity,
+        Vector3 predatorLocation, Entity predatorEntity, float speciesFocus, float speciesActivity,
+        float speciesAggression, float speciesFear, float strain, Random random)
     {
+        var ourCompounds = compoundStorage.Compounds;
         control.SetStateColonyAware(entity, MicrobeState.Normal);
+        MicrobeState predatorState = predatorEntity.Get<MicrobeControl>().State;
 
         ai.TargetPosition = (2 * (position.Position - predatorLocation)) + position.Position;
 
         control.LookAtPoint = ai.TargetPosition;
 
+        // Shouldn't this distance value scale with predator's and prey's cell radius?
+        // When the players cell is really big this distance value might not be enough
         if (position.Position.DistanceSquaredTo(predatorLocation) < 100.0f)
         {
             bool shouldSprint = true;
+
             if ((organelles.SlimeJets?.Count ?? 0) > 0 &&
                 RollCheck(speciesFear, Constants.MAX_SPECIES_FEAR, random))
             {
                 // There's a chance to jet away if we can
                 control.SecreteSlimeForSomeTime(ref organelles, random);
             }
+            else if (ourCompounds.GetCompoundAmount(Compound.Mucilage) > Constants.MUCOCYST_MINIMUM_MUCILAGE
+                     && organelles.MucocystCount > 0 && (strain >= Constants.MAX_STRAIN_PER_ENTITY * 0.70
```

Comment:
The preferred style in Thrive is to have the `&&` and `||` at the end of lines rather than at the beginning of lines.